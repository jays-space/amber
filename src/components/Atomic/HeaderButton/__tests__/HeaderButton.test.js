import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { HeaderButton } from "..";

describe("<HeaderButton />", () => {
  it("should render correctly", () => {
    render(<HeaderButton variant="primary" />);

    const headerButtonElement = screen.getByRole("button");
    const iconElement = screen.getByTestId("icon-add");

    expect(iconElement).toBeInTheDocument();
    expect(headerButtonElement).toBeInTheDocument();
    expect(headerButtonElement).toMatchSnapshot();
  });

  it("should render primary header button with icon", () => {
    render(<HeaderButton variant="primary" />);

    const primaryHeaderButtonElement = screen.getByTestId(
      "header-button-primary"
    );

    const iconElement = screen.getByTestId("icon-add");

    expect(iconElement).toBeInTheDocument();
    expect(primaryHeaderButtonElement).toBeInTheDocument();
  });

  it("should render secondary header button with icon", () => {
    render(<HeaderButton variant="secondary" />);

    const secondaryHeaderButtonElement = screen.getByTestId(
      "header-button-secondary"
    );

    const iconElement = screen.getByTestId("icon-notification");

    expect(iconElement).toBeInTheDocument();
    expect(secondaryHeaderButtonElement).toBeInTheDocument();
  });
});
