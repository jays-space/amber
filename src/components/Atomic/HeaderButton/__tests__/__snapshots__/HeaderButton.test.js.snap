// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderButton /> should render correctly 1`] = `
<button
  class="w-10 lg:w-14 aspect-square flex justify-center items-center rounded-full last-of-type:mx-4 cursor-pointer transform ease-out duration-100 cursor-pointer bg-yellow-400 hover:bg-opacity-80"
  data-testid="header-button-primary"
>
  <svg
    class="text-slate-500 !text-slate-800"
    data-testid="icon-add"
    fill="currentColor"
    height="28"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
    />
  </svg>
</button>
`;
